kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousClients.java

```diff
@@ -28,13 +30,22 @@
 class KroxyliciousClients {
     private final String bootstrapServers;
 
+    private final List<Admin> admins;
+    private final List<Producer> producers;
+    private final List<Consumer> consumers;
+
     KroxyliciousClients(String bootstrapServers) {
         this.bootstrapServers = bootstrapServers;
+        this.admins = new ArrayList<>();
+        this.producers = new ArrayList<>();
+        this.consumers = new ArrayList<>();
     }
 
     public Admin admin(Map<String, Object> additionalConfig) {
         Map<String, Object> config = createConfigMap(bootstrapServers, additionalConfig);
-        return CloseableAdmin.create(config);
+        Admin admin = CloseableAdmin.create(config);
+        admins.add(admin);
```

Comment:
Happy to leave it as is for this PR, but did we talk about sharing clients with the same config? Maybe thats a bad idea we should just let users create as many clients as they wish sharing them or not is a user problem?

Reply:
I think we decided comparing configs was a bad idea and could get a little hairy, but I'm happy to be corrected on that. At the moment, you can create one `KroxyliciousClients` per virtual cluster, but there's nothing to stop that from spawning infinite admins, etc. with the same config.

Reply:
yeah, thats cool. Lets leave it for now, we can revisit it later if we need to.